ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListSourceExecutor.java

```diff
@@ -191,14 +200,42 @@ private static SourceDescriptionWithWarnings describeSource(
       ), statementText);
     }
 
+    List<RunningQuery> sourceQueries = getQueries(ksqlEngine,
+        q -> q.getSourceNames().contains(dataSource.getName()));
+    List<RunningQuery> sinkQueries = getQueries(ksqlEngine,
+        q -> q.getSinkName().equals(dataSource.getName()));
+
     Optional<org.apache.kafka.clients.admin.TopicDescription> topicDescription = Optional.empty();
+    Optional<ConsumerGroupDescription> consumerGroupDescription = Optional.empty();
+    Map<TopicPartition, OffsetAndMetadata> topicAndConsumerOffsets = new LinkedHashMap<>();
+    Map<TopicPartition, ListOffsetsResultInfo> topicAndStartOffsets = new LinkedHashMap<>();
+    Map<TopicPartition, ListOffsetsResultInfo> topicAndEndOffsets = new LinkedHashMap<>();
     final List<KsqlWarning> warnings = new LinkedList<>();
     if (extended) {
       try {
         topicDescription = Optional.of(
             serviceContext.getTopicClient().describeTopic(dataSource.getKafkaTopicName())
         );
-      } catch (final KafkaException | KafkaResponseGetFailedException e) {
+        String serviceId = "default"; //FIXME not sure how to get this
+        if (sourceQueries.isEmpty()){
+          consumerGroupDescription = Optional.empty();
+        } else {
+          String queryId = sourceQueries.get(0).getId().toString();
+          String consumerGroupId = "_confluent-ksql-" + serviceId + "_" + KsqlConfig.KSQL_PERSISTENT_QUERY_NAME_PREFIX_DEFAULT + queryId; //FIXME there should be a better way to build this
```

Comment:
This is the only naive way I've found so far to get the consumer group id. Would love some feedback on how to get this in a proper/better way.

Reply:
I think this is the same as `StreamsConfig.APPLICATION_ID_CONFIG` in the `QueryMetadata.getStreamsProperties`, but I'd need to double check that

Reply:
You can see how we build it in `QueryExecutor#getApplicationId`